src/OSPSuite.R/MinimalImplementations/RGroupRepository.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using OSPSuite.Core.Domain;
+
+namespace OSPSuite.R.MinimalImplementations
+{
+   public class RGroupRepository : IGroupRepository
+   {
+      private readonly ConcurrentDictionary<string, IGroup> _allGroups = new ConcurrentDictionary<string, IGroup>();
+
+      public IEnumerable<IGroup> All() => _allGroups.Values;
+
+      public IGroup GroupByName(string groupName) => getOrAdd(groupName);
+
+      public IGroup GroupById(string groupId) => getOrAdd(groupId);
+
+      private IGroup getOrAdd(string groupIdOrName)
+      {
+         return _allGroups.GetOrAdd(groupIdOrName, x => new Group {Id = groupIdOrName, Name = groupIdOrName});
```

Comment:
Now I always have groups with same name and id. That way, all is consistent within the R run (id is what is used for serialization but name is what is used in the object model).